dev/src/gltexture.cpp

```diff
@@ -227,14 +242,20 @@ uint8_t *ReadTexture(const Texture &tex) {
     #endif
 }
 
-void SetImageTexture(int textureunit, const Texture &tex, int tf) {
+void SetImageTexture(int textureunit, const Texture &tex, int level, int tf) {
     #ifdef PLATFORM_WINNIX
-        if (glBindImageTexture)
-            GL_CALL(glBindImageTexture(textureunit, tex.id, 0, GL_TRUE, 0,
-                               tf & TF_WRITEONLY
+        if (glBindImageTexture) {
+            GLenum access = tf & TF_WRITEONLY
                                    ? GL_WRITE_ONLY
-                                   : (tf & TF_READWRITE ? GL_READ_WRITE : GL_READ_ONLY),
-                               tf & TF_FLOAT ? GL_RGBA32F : GL_RGBA8));
+                                   : (tf & TF_READWRITE ? GL_READ_WRITE : GL_READ_ONLY);
+            // Handle format
+            GLenum format = tf & TF_SINGLE_CHANNEL ? GL_R8 : GL_RGBA8;
```

Comment:
we have tf to format conversion also elsewhere in the code, is it the same? maybe lift into function?

Reply:
Ideally this would be refactored into also allowing more than 1 or 4 channel formats?

Reply:
Once we need it, yes.
In this case, I am just talking about copy-paste code, which we generally want to avoid